lib/msal-node/src/client/ManagedIdentitySources/AzureArc.ts

```diff
@@ -3,17 +3,181 @@
+} from "../../error/ManagedIdentityError";
+import {
+    AUTHORIZATION_HEADER_NAME,
+    HttpMethod,
+    METADATA_HEADER_NAME,
+    ManagedIdentityIdType,
+} from "../../utils/Constants";
+import { NodeStorage } from "../../cache/NodeStorage";
+import * as fs from "fs";
+import { ManagedIdentityTokenResponse } from "../../response/ManagedIdentityTokenResponse";
 
+const ARC_API_VERSION: string = "2019-11-01";
+
+/**
+ * Original source of code: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/AzureArcManagedIdentitySource.cs
+ */
 export class AzureArc extends BaseManagedIdentitySource {
+    private endpoint: string;
+
+    constructor(
+        logger: Logger,
+        nodeStorage: NodeStorage,
+        networkClient: INetworkModule,
+        cryptoProvider: CryptoProvider,
+        endpoint: string
+    ) {
+        super(logger, nodeStorage, networkClient, cryptoProvider);
+
+        this.endpoint = endpoint;
+    }
+
+    public static tryCreate(
+        logger: Logger,
+        nodeStorage: NodeStorage,
+        networkClient: INetworkModule,
+        cryptoProvider: CryptoProvider
+    ): AzureArc | null {
+        const imdsEndpoint: string | undefined = process.env["IMDS_ENDPOINT"];
+
+        const [areEnvironmentVariablesValidated, endpoint]: [
+            boolean,
+            string | undefined
+        ] = validateEnvironmentVariables(
+            process.env["IDENTITY_ENDPOINT"] || undefined,
+            imdsEndpoint,
+            logger
+        );
+
+        return areEnvironmentVariablesValidated
+            ? new AzureArc(
+                  logger,
+                  nodeStorage,
+                  networkClient,
+                  cryptoProvider,
+                  endpoint as string
+              )
+            : null;
+    }
+
     public createRequest(
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        _resource: string,
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        _managedIdentityId: ManagedIdentityId
+        resource: string,
+        managedIdentityId: ManagedIdentityId
     ): ManagedIdentityRequestParameters {
-        throw new Error("Method not implemented.");
+        if (
+            managedIdentityId.idType !== ManagedIdentityIdType.SYSTEM_ASSIGNED
+        ) {
+            throw createManagedIdentityError(
+                ManagedIdentityErrorCodes.unableToCreateAzureArc
+            );
+        }
+
+        const request: ManagedIdentityRequestParameters =
+            new ManagedIdentityRequestParameters(HttpMethod.GET, this.endpoint);
+
+        request.headers[METADATA_HEADER_NAME] = "true";
+        request.queryParameters["api-version"] = ARC_API_VERSION;
+        request.queryParameters["resource"] = resource;
+        // bodyParameters calculated in BaseManagedIdentity.acquireTokenWithManagedIdentity
+
+        return request;
+    }
+
+    public async retryPolicy(
```

Comment:
This method should not be named retryPolicy. The concept of retry policy is different where you retry for certain error codes. This is handling the response and making the request with authHeader. In MSAL .Net and Java this is implemented by overriding the `HandleResponse` method from the base class and adding the logic of extracting the auth header from the response and making the request again.

Try to use a similar pattern.